lib_pypy/_sqlite3.py

```
@@ -441,16 +536,26 @@ def __init__(self, database, timeout=5.0, detect_types=0, isolation_level="",
         self.DataError = DataError
         self.NotSupportedError = NotSupportedError
         sys.audit("sqlite3.connect/handle", self)
+        # CPython 3.14 `pysqlite_connection_init_impl`: disabled autocommit
+        # starts and thereafter continuously maintains an explicit transaction.
+        if self._autocommit is False:
+            self._begin()
 
     def __del__(self):
         if self._db:
+            # CPython 3.14 `connection_finalize` reports an unclosed database.
+            warnings.warn("unclosed database in %r" % self,
```

Ruff (0.16.2)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)